client/app/components/packages/save-button.js

```diff
@@ -19,7 +19,7 @@ export default class SaveButtonComponent extends Component {
 
   // triggered when the button is clicked
   @task(function* () {
-    yield this.args.onClick();
-  })
+    yield this.args._saveChangeset(this.args.onClick);
```

Comment:
Does this assume that if we're reusing the `save-button` component in different forms that every attribute that's saveable is also associated with a changeset? Just thinking about attributes that don't have any validation requirements on them for saving or submitting. I know then we'd probably just pass in the model rather than `saveableForm.saveableChanges.attribute_name`, but if this action is stored in the `save-button` component are we always expecting a changeset?

Reply:
Yes you're right, but here I'm making save-button very much a dependency on saveable-form. So that's the tradeoff: save-button because something more specific to saveable-form.

I'll try to think of a way to "curry" this in a more agnostic way... 

Great question. It may just be that the naming is wrong, and should not refer to changeset. It behaves as an optional wrapper function around the public onClick. 

Going to think about it some more...